src/imitation/algorithms/mce_irl.py

```diff
@@ -135,6 +136,50 @@ def squeeze_r(r_output: th.Tensor) -> th.Tensor:
     return r_output
 
 
+# TODO(adam): add test case for this
+class TabularPolicy(policies.BasePolicy):
+    """A tabular policy. Cannot be trained -- prediction only."""
+
+    def __init__(self, pi: np.ndarray, rng: Optional[np.random.RandomState]):
+        """Builds TabularPolicy.
+
+        Args:
+            pi: A tabular policy. Three-dimensional array, where pi[t,s,a]
+                is the probability of taking action a at state s at timestep t.
+            rng: Random state, used for sampling when `predict` is called with
+                `deterministic=False`.
+        """
+        super().__init__()
+        assert pi.ndim == 3, "expected three-dimensional policy"
+        assert np.allclose(pi.sum(axis=2), 1), "policy not normalized"
+        assert np.all(pi >= 0), "policy has negative probabilities"
+        self.pi = pi
+        self.rng = rng or np.random
+
+    def predict(
+        self,
+        observation: np.ndarray,
+        state: Optional[np.ndarray] = None,
```

Comment:
Thanks for flagging this discrepancy. I think I might just be confused about what tabular MDPs observation is meant to be, then. In a POMDP, the policy would see the observation; the reward is typically a function of state, but for a learned reward it has to be a function of observation. Sounds like in the current implementation this is swapped: policy sees state but reward sees observation?

Reply:
Yes. This is standard for MCE IRL: the "observations" only exist so you can get the benefits of function approximation for your reward function (namely, it forces semantically similar states to have similar reward). Given that you are doing planning to obtain your policy anyway, there is no need for function approximation in the policy.

Reply:
Thanks for the explanation. We need to stick to the SB3 naming convention for this function for compatibility with any callees that are using keyword arguments. But I've added a discussion of this in the docstring to avoid confusion.